# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/base/nsURLParsers.cpp
# Commit: 9ad299b5161b
# Full Hash: 9ad299b5161b0cedf852d56365310de499f94043
# Author: Valentin Gosu <valentin.gosu@gmail.com>
# Date: 2018-11-14 22:12:17
# Regressor Bug: 1495313
# File Overlap Count: 1
# Description:
#   Bug 1495313 - Allow empty username in URLs r=dragana
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D11254
# ==============================================================================

diff -r 6293789f6e9c -r 9ad299b5161b netwerk/base/nsURLParsers.cpp
--- a/netwerk/base/nsURLParsers.cpp	Wed Nov 14 14:27:58 2018 +0000
+++ b/netwerk/base/nsURLParsers.cpp	Wed Nov 14 15:24:21 2018 +0000
@@ -535,10 +535,6 @@
     const char *p = (const char *) memchr(userinfo, ':', userinfoLen);
     if (p) {
         // userinfo = <username:password>
-        if (p == userinfo) {
-            // must have a username!
-            return NS_ERROR_MALFORMED_URI;
-        }
         SET_RESULT(username, 0, p - userinfo);
         SET_RESULT(password, p - userinfo + 1, userinfoLen - (p - userinfo + 1));
     }